docs/04_resource_management/041_exception_handling.ipynb

```diff
@@ -0,0 +1,155 @@
+    "\n",
+    "Wenn Fehler erkannt werden, dann werden Fehler (in der Programmierung meistens Ausnahmen) geworfen (auf Englisch \"raise exceptions\").\n",
+    "\n",
+    "Fehler treten z.B. auf, wenn ein unerwarteter Input geliefert wird, oder wenn versucht wird, eine Zahl durch 0 zu teilen.\n",
+    "\n",
+    "Wenn wir wissen, dass ein bestimmter Code bestimmte Fehler werfen kann und wir diese Fehler abfangen wollen, dann können wir das mit den Keywords `try` und `except` tun:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "try:\n",
+    "    number = int(input(\"Please give a number: \"))\n",
+    "    print(f\"100 / {number} = {100/number}\")\n",
+    "except Exception as exception:\n",
+    "    print(type(exception))\n",
+    "    print(exception)"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Versuche beim obigen Code Fehler zu provozieren.\n",
+    "\n",
+    "Grundsätzlich gibt es bei diesem Code zwei häufige Fehlerquellen:\n",
+    "1. Der User gibt nicht eine Zahl sondern sonst etwas ein -> die Eingabe kann nicht in eine Zahl umgewandelt werden.\n",
+    "   \n",
+    "   Dies führt zu folgender Ausgabe:\n",
+    "    ```\n",
+    "    <class 'ValueError'>\n",
+    "    invalid literal for int() with base 10: 'ungültige Zahl'\n",
+    "    ```\n",
+    "2. Der User gibt die Zahl 0 ein -> dies führt zu einer 0-Division.\n",
+    "   \n",
+    "   Dies führt zu dieser Ausgabe:\n",
+    "   ```\n",
+    "   <class 'ZeroDivisionError'>\n",
+    "   division by zero\n",
+    "   ```\n",
+    "\n",
+    "Oft wirst du in die Situation kommen, dass du verschiedene Fehler verschieden oder nur bestimmte Fehler behandeln möchtest.\n",
+    "Hierfür bietet es sich an, verschiedene `except`-Blöcke zu definieren:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "try:\n",
+    "    number = int(input(\"Please give a number: \"))\n",
+    "    print(f\"100 / {number} = {100/number}\")\n",
+    "except ZeroDivisionError:\n",
+    "    print(\"Use another input than 0. Can't divide by 0.\")\n",
+    "except ValueError:\n",
+    "    print(\"Please enter a valid number!\")"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Nach dem `except`-Keyword haben wir den Typ der Exception (des Fehlers) angegeben. Der Inhalt des entsprechenden `except`-Blocks wird nur ausgeführt, wenn der Fehler vom spezifizierten Typ ist (bzw. davon erbt).\n",
+    "\n",
+    "Im allerersten `except` haben wir die Exception-Klasse `Exception` angegeben. Da die anderen Exception-Klassen wie `ZeroDivisionError` und `ValueError` eine Subklasse von `Exception` sind, werden diese Fehler auch von `except Exception` abgefangen.\n",
```

Comment:
Ich würde hier noch folgendes erwähnen:
- Wenn 'except' ohne Exception-Class verwendet wird, wird der Block bei jede Exception ausgeführt
- Wenn man exakte Exceptions abfängt, soll am Schluss noch einen Block für alle andere Exceptions hinzugefügt werden, damit auch solche "graceful" abgefangen und bearbeitet werden.

Reply:
Für ersten Punkt wurde ein neuer Abschnitt hinzugefügt.
Für den zweiten: habe einen weiteren Bullet-Point in "Vollständiges Beispiel" hinzugefügt.